src/poetry/installation/chef.py

```diff
@@ -17,12 +32,175 @@
+            env = IsolatedEnv(venv, self._config)
+            builder = ProjectBuilder(
+                directory,
+                python_executable=env.executable,
+                scripts_dir=env.scripts_dir,
+                runner=quiet_subprocess_runner,
+            )
+            env.install(builder.build_system_requires)
+            env.install(
+                builder.build_system_requires | builder.get_requires_for_build("wheel")
+            )
+
+            stdout = StringIO()
+            error: Exception | None = None
+            try:
+                with redirect_stdout(stdout):
+                    path = Path(
+                        builder.build(
+                            "wheel" if not editable else "editable",
+                            destination.as_posix(),
+                        )
+                    )
+            except BuildBackendException as e:
+                if isinstance(e.exception, CalledProcessError) and (
+                    e.exception.stdout is not None or e.exception.stderr is not None
+                ):
+                    message = (
+                        e.exception.stderr.decode()
+                        if e.exception.stderr is not None
+                        else e.exception.stdout.decode()
+                    )
+                    error = ChefBuildError(message)
+                else:
+                    error = ChefBuildError(str(e))
+
+            if error is not None:
+                raise error from None
+
+            return path
+
+    def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:
+        from poetry.core.packages.utils.link import Link
+
+        suffix = archive.suffix
+        context: Callable[
+            [str], AbstractContextManager[zipfile.ZipFile | tarfile.TarFile]
+        ]
+        if suffix == ".zip":
+            context = zipfile.ZipFile
+        else:
+            context = tarfile.open
+
+        with temporary_directory() as tmp_dir:
+            with context(archive.as_posix()) as archive_archive:
+                archive_archive.extractall(tmp_dir)
+
+            archive_dir = Path(tmp_dir)
+
+            elements = list(archive_dir.glob("*"))
+
+            if len(elements) == 1 and elements[0].is_dir():
+                sdist_dir = elements[0]
+            else:
+                sdist_dir = archive_dir / archive.name.rstrip(suffix)
+                if not sdist_dir.is_dir():
+                    sdist_dir = archive_dir
+
+            if destination is None:
+                destination = self.get_cache_directory_for_link(Link(archive.as_uri()))
+
+            destination.mkdir(parents=True, exist_ok=True)
+
+            return self._prepare(
+                sdist_dir,
+                destination,
+            )
+
+    def should_prepare(self, archive: Path) -> bool:
+        return archive.is_dir() or not self.is_wheel(archive)
+
+    @classmethod
+    def is_wheel(cls, archive: Path) -> bool:
```

Comment:
```suggestion
    def _is_wheel(cls, archive: Path) -> bool:
```
only used internally